encord/http/v2/api_client.py

```diff
@@ -79,6 +81,25 @@ def _headers(self):
             HEADER_CLOUD_TRACE_CONTEXT: self._tracing_id(),
         }
 
+    def get_bound_partial(self, callback: Callable) -> Callable:
```

Comment:
I think the name is a bit vague here. How about?
```suggestion
    def ensure_bound_api_client(self, callback: Callable) -> Callable:
```

Reply:
well it's kind of "`functools.partial` with a cache", and the cache has to live in this object to prevent the memory leak/lifecycle breakage.

Not sure either name adequately describes this; I like mine better because the point is to *create* a function. `ensure_xyz` reads more like an assertion to me, which is not what I'm doing there.

One thing this reminds me of is that I should just `raise RuntimeError` if the `api_client` is missing from the signature instead of "silently do nothing"

Reply:
renamed to `get_bound_operation`, as it's not a 'callback' really.